alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteMonitorAction.kt

```diff
@@ -140,68 +139,102 @@ class TransportDeleteMonitorAction @Inject constructor(
             )
             return ScheduledJob.parse(xcp, getResponse.id, getResponse.version) as Monitor
         }
+    }
+
+    companion object {
+        @JvmStatic
+        suspend fun deleteAllResourcesForMonitor(
+            client: Client,
+            monitor: Monitor,
+            deleteRequest: DeleteRequest,
+            monitorId: String,
+        ): DeleteResponse {
+            val deleteResponse = deleteMonitorDocument(client, deleteRequest)
+            deleteMetadata(client, monitor)
+            deleteDocLevelMonitorQueriesAndIndices(client, monitor, monitorId)
+            return deleteResponse
+        }
 
-        private suspend fun deleteMonitor(monitor: Monitor): DeleteResponse {
+        private suspend fun deleteMonitorDocument(client: Client, deleteRequest: DeleteRequest): DeleteResponse {
             return client.suspendUntil { delete(deleteRequest, it) }
         }
 
-        private suspend fun deleteMetadata(monitor: Monitor) {
+        suspend fun deleteMetadata(client: Client, monitor: Monitor) {
             val deleteRequest = DeleteRequest(ScheduledJob.SCHEDULED_JOBS_INDEX, "${monitor.id}-metadata")
-            val deleteResponse: DeleteResponse = client.suspendUntil { delete(deleteRequest, it) }
+                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            try {
+                val deleteResponse: DeleteResponse = client.suspendUntil { delete(deleteRequest, it) }
+                log.debug("Monitor metadata: ${deleteResponse.id} deletion result: ${deleteResponse.result}")
+            } catch (e: Exception) {
+                // we only log the error and don't fail the request because if monitor document has been deleted,
+                // we cannot retry based on this failure
+                log.error("Failed to delete workflow metadata for monitor ${monitor.id}.", e)
```

Comment:
Lets put the workflow metadata id in here, so it gives us the ability to delete it manually if needed.